src/kontrol/utils.py

```diff
@@ -716,3 +716,137 @@ def decode_log_message(token: str, selector: int) -> str | None:
     238520724: ['address', 'address', 'address', 'bool'],
     1717301556: ['address', 'address', 'address', 'address'],
 }
+
+
+def kontrol_test_file_contents() -> str:
+    return """pragma solidity ^0.8.26;
+import "forge-std/Vm.sol";
+import "forge-std/Test.sol";
+import "kontrol-cheatcodes/KontrolCheats.sol";
+
+contract KontrolTest is Test, KontrolCheats {
+    enum Mode {
+        Assume,
+        Try,
+        Assert
+    }
+    function _establish(Mode mode, bool condition) internal pure returns (bool) {
+        if (mode == Mode.Assume) {
+            vm.assume(condition);
+            return true;
+        } else if (mode == Mode.Try) {
+            return condition;
+        } else {
+            assert(condition);
+            return true;
+        }
+    }
+    function _loadData(
+        address contractAddress,
+        uint256 slot,
+        uint256 offset,
+        uint256 width
+    ) internal view returns (uint256) {
+        // `offset` and `width` must not overflow the slot
+        assert(offset + width <= 32);
+        // Slot read mask
+        uint256 mask;
+        unchecked {
+            mask = (2 ** (8 * width)) - 1;
+        }
```

Comment:
Some additional validation: width should be >0, address should be non-zero;
Additionally, `mask = (2 ** (8 * width)) - 1;` should overflow if `width` is 32 bytes.

```suggestion
    function _loadData(
        address contractAddress,
        uint256 slot,
        uint256 offset,
        uint256 width
    ) internal view returns (uint256) {
        require(contractAddress != address(0), 'Invalid contract address');
        require(width > 0, 'Width must be greater than 0');
        // `offset` and `width` must not overflow the slot
        assert(offset + width <= 32);
        
        // Slot read mask - handle width = 32 case to prevent overflow
        uint256 mask;
        if (width == 32) {
            mask = type(uint256).max;
        } else {
            unchecked {
                mask = (2 ** (8 * width)) - 1;
            }
        }
```

Reply:
thanks!